.github/workflows/ci.yml

```diff
@@ -0,0 +1,126 @@
+name: CI Workflow
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - rel/**
+  pull_request:
+    branches:
+      - main
+
+permissions: read-all
+
+jobs:
+  lint:
+    name: Linting
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10.14'
+
+      - name: Install dev dependencies
+        run: pip install -r requirements-dev.txt
+
+      - name: Run pre-commit linting
+        run: pre-commit run --all-files
+
+  system-tests:
+    name: System Tests
+    runs-on: windows-latest
+    needs: [tests,lint]
+    if: github.event_name != 'pull_request' && (github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/rel/'))
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10.8'
```

Comment:
Do we need to specify the patch version? If not, I would go with `3.10`, if yes, I would use the latest `3.10.16` (also above for "Linting")